.github/workflows/pr-deploy.yaml

```diff
@@ -318,6 +309,90 @@ jobs:
           kubectl create secret generic coder-db-url -n pr${{ env.PR_NUMBER }} \
             --from-literal=url="postgres://coder:coder@coder-db-postgresql.pr${{ env.PR_NUMBER }}.svc.cluster.local:5432/coder?sslmode=disable"
 
+      - name: Create a kubeconfig for the workspace
+        if: needs.get_info.outputs.NEW == 'true'
+        # This service account will be used to grant full access to the namespace from the workspace
+        run: |
+          set -e pipefail
+          # Create service account, role, rolebinding and secret
+          cat <<EOF | kubectl apply -f -
```

Comment:
This is very unwieldy, and should probably go into a file. You could use `envsubst` to fill in the dynamic values in the file with env vars